pandas/tests/arrays/test_integer.py

```diff
@@ -710,6 +710,22 @@ def test_to_integer_array_float():
     assert result.dtype == Int64Dtype()
 
 
+@pytest.mark.parametrize(
+    "values, dtype",
+    [
+        ([9999999999999999, 123123123123123123, 10000000000000543, np.nan], "Int64"),
+        (np.array([9999999999999999, 123123123123123123, 10000000000000543],
+         dtype=np.int64), "Int64"),
+    ],
+)
+def test_to_integer_array_large_numbers(values, dtype):
+    result = integer_array(values, dtype=dtype)
+    assert result.dtype == dtype
+    for i in range(len(values)):
+        if not np.isnan(values[i]):
+            assert result[i] == values[i]
```

Comment:
Can you use `tm.assert_series_equal(result, expected)`. This pattern is used extensively. look at other tests for examples.

Reply:
#30347 needs to be fixed first.

Reply:
Oh I didn't know about `check_exact`.
I have updated it.